words2num/lang_EN_US.py

```diff
@@ -135,6 +146,19 @@ def tokenize(text):
         decimal = False
         parsed_tokens = []
         decimal_tokens = []
+        mul_tokens = []
+        pvs = compute_pvs(tokens)
+        # Remove multiplier tokens from the tokens list
+        # These will be used to multiply the computed value before returning
+        while max(pvs) == pvs[-1] and len(pvs) > 1 and max(pvs) > 1:
+            # Three conditions must be met:
+            # 1: The last token in the list must have the highest placevalue of any token
+            # 2: The list of tokens must be longer than one 
+                # (to prevent extracting all tokens as mul_tokens)
+            # 3: The maximum placevalue must be greater than 1
+                # This limits our mul_tokens to "hundred" or greater
+            mul_tokens.insert(0, VOCAB[tokens.pop()])
```

Comment:
Doesn't this pop nullify the iteration over tokens below?

Reply:
The goal is to remove the "multiplier" tokens before going through the rest of the logic, since they don't follow conventional number structure. 

The example in the comment above is a decent illustration of what I mean by conventional number structure. Typically you have groupings of numbers by decreasing value (`six million` `four hundred thousand` `five`) but when the last number in a sentence is the largest place value it breaks some of the logic we're using. This is mostly a problem when we have decimal numbers followed by a modifier